src/diffusers/pipelines/deepfloyd_if/pipeline_if_superresolution.py

```diff
@@ -543,12 +543,27 @@ def check_inputs(
 
         if isinstance(image, list):
             image_batch_size = len(image)
+            # Check that each image is the same size:
```

Comment:
I think it is better to do this in a separate code block:
So we keep this section as it is to check `image_batch_size`

and then
```
if isinstance(image, list):
    check_image_size = image[0]
else:
    check_image_size = image

if isinstance(check_image_size, PIL.Image.Image):
    image_size = check_image_size.size
elif isinstance(check_image_size, torch.Tensor):
   image_size = check_image_size.shape[2:]
elif isinstanc(..., np.ndarray):
    image_size = check_image.shape[:1]

if image_size ....:
    raise ValueError(...)

```

The current code does not work for list of array or tensors